src/components/application_manager/src/commands/command_impl.cc

```diff
@@ -114,7 +114,9 @@ bool CommandImpl::ReplaceMobileWithHMIAppId(
         smart_objects::SmartArray* message_array = message.asArray();
         smart_objects::SmartArray::iterator it = message_array->begin();
         for (; it != message_array->end(); ++it) {
-          ReplaceMobileWithHMIAppId(*it);
+          if (!ReplaceMobileWithHMIAppId(*it)) {
+            break;
```

Comment:
break  cause  the function return true, however, it shall return false.

Reply:
@yang1070 
Done with 866dfe8

Reply:
@yang1070 
Reverted previous changes due to issues with ATF tests. As I found there is a mixed logic for two use cases: first is related to bugfix done for Ford Sync transport switch implementation which requires checking of application with app_id, second is common usage when app_id is replaced during message processing. So I propose to leave code as is for now and create another issue for investigation how this part can be re-factored.